File: app/src/main/java/com/schepor/gita/presentation/home/LessonNodeComponents.kt
package com.schepor.gita.presentation.home

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.schepor.gita.presentation.theme.Spacing

/**
 * Section Header - Displays "SECTION X, UNIT Y" style header
 */
@Composable
fun SectionHeader(
    sectionNumber: Int,
    unitNumber: Int,
    description: String,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = Spacing.space8),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.primaryContainer
        )
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.space16)
        ) {
            Text(
                text = "SECTION $sectionNumber, UNIT $unitNumber",
                style = MaterialTheme.typography.labelLarge,
                fontWeight = FontWeight.Bold,
                color = MaterialTheme.colorScheme.primary
            )
            Spacer(modifier = Modifier.height(Spacing.space4))
            Text(
                text = description,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onPrimaryContainer
            )
        }
    }
}

/**
 * Lesson Node - Circular button representing a lesson
 */
@Composable
fun LessonNode(
    lessonNumber: Int,
    isUnlocked: Boolean,
    isCompleted: Boolean,
    isCurrent: Boolean,
    description: String = "",
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // The circular lesson button
        Box(
            modifier = Modifier
                .size(if (isCurrent) 80.dp else 64.dp)
                .shadow(
                    elevation = if (isCurrent) 8.dp else 0.dp,
                    shape = CircleShape
                )
                .background(
                    color = when {
                        !isUnlocked -> MaterialTheme.colorScheme.surfaceVariant
                        isCurrent -> Color(0xFF66BB6A) // Bright green for current
                        isCompleted -> Color(0xFF4CAF50) // Green for completed
                        else -> Color(0xFF4CAF50) // Green for unlocked
                    },
                    shape = CircleShape
                )
                .border(
                    width = if (isCurrent) 4.dp else 0.dp,
                    color = if (isCurrent) Color(0xFF81C784) else Color.Transparent,
                    shape = CircleShape
                )
                .clickable(enabled = isUnlocked, onClick = onClick),
            contentAlignment = Alignment.Center
        ) {
            when {
                !isUnlocked -> {
                    Icon(
                        imageVector = Icons.Default.Lock,
                        contentDescription = "Locked",
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(32.dp)
                    )
                }
                isCompleted -> {
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Completed",
                        tint = Color.White,
                        modifier = Modifier.size(40.dp)
                    )
                }
                else -> {
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Available",
                        tint = Color.White.copy(alpha = 0.5f),
                        modifier = Modifier.size(40.dp)
                    )
                }
            }
        }
        
        // Description text below the node
        if (description.isNotEmpty()) {
            Spacer(modifier = Modifier.height(Spacing.space8))
            Text(
                text = description,
                style = MaterialTheme.typography.bodyMedium,
                color = if (isUnlocked) 
                    MaterialTheme.colorScheme.onSurface 
                else 
                    MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.6f),
                textAlign = TextAlign.Center,
                modifier = Modifier.widthIn(max = 120.dp)
            )
        }
    }
}

/**
 * Progress Path - Vertical line connecting lesson nodes
 */
@Composable
fun ProgressPath(
    isUnlocked: Boolean,
    height: Int = 40,
    modifier: Modifier = Modifier
) {
    Canvas(
        modifier = modifier
            .width(4.dp)
            .height(height.dp)
    ) {
        drawLine(
            color = if (isUnlocked) Color(0xFFE0E0E0) else Color(0xFFE0E0E0).copy(alpha = 0.3f),
            start = Offset(size.width / 2, 0f),
            end = Offset(size.width / 2, size.height),
            strokeWidth = size.width
        )
    }
}

/**
 * Chapter Milestone - Special markers like treasure chests, characters, trophies
 */
@Composable
fun ChapterMilestone(
    type: MilestoneType,
    isUnlocked: Boolean,
    onClick: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .size(72.dp)
            .background(
                color = if (isUnlocked) 
                    MaterialTheme.colorScheme.tertiaryContainer 
                else 
                    MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f),
                shape = CircleShape
            )
            .then(
                if (onClick != null && isUnlocked) {
                    Modifier.clickable(onClick = onClick)
                } else {
                    Modifier
                }
            ),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = when (type) {
                MilestoneType.TREASURE -> "🎁"
                MilestoneType.CHARACTER -> "🦜"
                MilestoneType.TROPHY -> "🏆"
                MilestoneType.KRISHNA -> "🪷"
            },
            style = MaterialTheme.typography.displaySmall,
            modifier = Modifier.size(48.dp)
        )
    }
}

/**
 * Types of milestones that can appear in the progression
 */
enum class MilestoneType {
    TREASURE,   // Completion rewards
    CHARACTER,  // Krishna mascot checkpoints
    TROPHY,     // Chapter completion
    KRISHNA     // Special Krishna wisdom points
}

File: gita_flutter/lib/presentation/components/lesson_node_components.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_theme.dart';

/// Types of milestones in the lesson progression
enum MilestoneType {
  treasure,
  character,
  krishna,
  trophy,
}

/// Section header for chapter groupings - Duolingo style
class SectionHeader extends StatelessWidget {
  final int sectionNumber;
  final int unitNumber;
  final String description;
  final VoidCallback? onInfoTap;

  const SectionHeader({
    super.key,
    required this.sectionNumber,
    required this.unitNumber,
    required this.description,
    this.onInfoTap,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.space16),
      padding: const EdgeInsets.symmetric(
        horizontal: Spacing.space16,
        vertical: Spacing.space12,
      ),
      decoration: BoxDecoration(
        color: Theme.of(context).colorScheme.primary,
        borderRadius: BorderRadius.circular(16),
      ),
      child: Row(
        children: [
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  'SECTION $sectionNumber, UNIT $unitNumber',
                  style: Theme.of(context).textTheme.labelSmall?.copyWith(
                    color: Theme.of(context).colorScheme.onPrimary.withOpacity(0.9),
                    letterSpacing: 1.0,
                    fontWeight: FontWeight.w600,
                  ),
                ),
                const SizedBox(height: Spacing.space4),
                Text(
                  description,
                  style: Theme.of(context).textTheme.titleMedium?.copyWith(
                    color: Theme.of(context).colorScheme.onPrimary,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ],
            ),
          ),
          // Shloka info icon
          Container(
            width: 40,
            height: 40,
            decoration: BoxDecoration(
              color: Theme.of(context).colorScheme.onPrimary.withOpacity(0.2),
              borderRadius: BorderRadius.circular(8),
            ),
            child: IconButton(
              icon: Icon(
                Icons.menu_book_rounded,
                color: Theme.of(context).colorScheme.onPrimary,
                size: 22,
              ),
              onPressed: onInfoTap,
              tooltip: 'View shlokas in this unit',
            ),
          ),
        ],
      ),
    );
  }
}

/// Lesson node in the progression tree
class LessonNode extends StatelessWidget {
  final int lessonNumber;
  final bool isUnlocked;
  final bool isCompleted;
  final bool isCurrent;
  final String description;
  final VoidCallback onTap;

  const LessonNode({
    super.key,
    required this.lessonNumber,
    required this.isUnlocked,
    required this.isCompleted,
    required this.isCurrent,
    required this.description,
    required this.onTap,
  });

  Color _getBackgroundColor(BuildContext context) {
    if (isCompleted) {
      return Theme.of(context).colorScheme.primary;
    } else if (isCurrent) {
      return Theme.of(context).colorScheme.secondary;
    } else if (isUnlocked) {
      return Theme.of(context).colorScheme.surfaceContainerHighest;
    } else {
      return Theme.of(context).colorScheme.outline;
    }
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        GestureDetector(
          onTap: isUnlocked ? onTap : null,
          child: AnimatedContainer(
            duration: const Duration(milliseconds: 300),
            width: isCurrent ? 80 : 64,
            height: isCurrent ? 80 : 64,
            decoration: BoxDecoration(
              color: _getBackgroundColor(context),
              shape: BoxShape.circle,
              border: isCurrent
                  ? Border.all(
                      color: Theme.of(context).colorScheme.primary,
                      width: 4,
                    )
                  : null,
              boxShadow: isCurrent
                  ? [
                      BoxShadow(
                        color: Theme.of(context)
                            .colorScheme
                            .primary
                            .withOpacity(0.4),
                        blurRadius: 16,
                        spreadRadius: 2,
                      ),
                    ]
                  : null,
            ),
            child: Center(
              child: isCompleted
                  ? Icon(
                      Icons.check,
                      color: Theme.of(context).colorScheme.onPrimary,
                      size: 32,
                    )
                  : isUnlocked
                      ? Text(
                          '$lessonNumber',
                          style: Theme.of(context)
                              .textTheme
                              .titleLarge
                              ?.copyWith(
                                color: isCompleted || isCurrent
                                    ? Theme.of(context).colorScheme.onPrimary
                                    : Theme.of(context).colorScheme.onSurface,
                                fontWeight: FontWeight.bold,
                              ),
                        )
                      : Icon(
                          Icons.lock,
                          color: Theme.of(context).colorScheme.onSurface.withOpacity(0.5),
                          size: 24,
                        ),
            ),
          ),
        ),
        if (description.isNotEmpty && (isCurrent || isCompleted))
          Padding(
            padding: const EdgeInsets.only(top: Spacing.space8),
            child: Text(
              description,
              style: Theme.of(context).textTheme.bodySmall,
              textAlign: TextAlign.center,
            ),
          ),
      ],
    );
  }
}

/// Progress path connecting lesson nodes
class ProgressPath extends StatelessWidget {
  final bool isUnlocked;
  final double height;

  const ProgressPath({
    super.key,
    required this.isUnlocked,
    this.height = 32,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 8,
      height: height,
      decoration: BoxDecoration(
        color: isUnlocked
            ? Theme.of(context).colorScheme.primary
            : Theme.of(context).colorScheme.outline,
        borderRadius: BorderRadius.circular(4),
      ),
    );
  }
}

/// Chapter milestone (treasure, character, trophy)
class ChapterMilestone extends StatelessWidget {
  final MilestoneType type;
  final bool isUnlocked;

  const ChapterMilestone({
    super.key,
    required this.type,
    required this.isUnlocked,
  });

  String _getEmoji() {
    switch (type) {
      case MilestoneType.treasure:
        return '📦';
      case MilestoneType.character:
        return '🧘';
      case MilestoneType.krishna:
        return '🪈';
      case MilestoneType.trophy:
        return '🏆';
    }
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 56,
      height: 56,
      decoration: BoxDecoration(
        color: isUnlocked
            ? const Color(0xFFFFD700)
            : Theme.of(context).colorScheme.surfaceContainerHighest,
        shape: BoxShape.circle,
        boxShadow: isUnlocked
            ? [
                BoxShadow(
                  color: const Color(0xFFFFD700).withOpacity(0.4),
                  blurRadius: 12,
                  spreadRadius: 2,
                ),
              ]
            : null,
      ),
      child: Center(
        child: Text(
          _getEmoji(),
          style: TextStyle(
            fontSize: 28,
            color: isUnlocked ? null : Colors.grey,
          ),
        ),
      ),
    );
  }
}
